pkg/awsutils/awsutils.go

```diff
@@ -392,15 +409,56 @@ func (cache *EC2InstanceMetadataCache) refreshSGIDs(mac string) error {
 	newSGs := StringSet{}
 	newSGs.Set(sgIDs)
 	addedSGs := newSGs.Difference(&cache.securityGroups)
+	addedSGsCount := 0
 	deletedSGs := cache.securityGroups.Difference(&newSGs)
+	deletedSGsCount := 0
 
 	for _, sg := range addedSGs.SortedList() {
 		log.Infof("Found %s, added to ipamd cache", sg)
+		addedSGsCount++
 	}
 	for _, sg := range deletedSGs.SortedList() {
 		log.Infof("Removed %s from ipamd cache", sg)
+		deletedSGsCount++
 	}
 	cache.securityGroups.Set(sgIDs)
+
+	if (cache.useCustomNetworking == false) && (addedSGsCount != 0 || deletedSGsCount != 0) {
```

Comment:
Nit: `if !cache.useCustomNetworking && (addedSGsCount != 0 || deletedSGsCount != 0) {` would be a bit more "go like"

Reply:
sure will fix it and commit :D